Add complementary BUILD_WEEK and BUILDS_THIS_WEEK

adamwasila opened the pull request:
Useful for projects that have constant weekly increments. Allows to use build numbering scheme that have current week of the year number and consecutive number, e.g.: "wk10.1, wk10.2, wk10.3, wk11.1, wk11.2, ...". This is supplementary to day/month/year periods.

Bagira80 commented:
Could you please update your pull-request to work with the changes in master that have been applied since March?

adamwasila commented:
Yes, I will! Probably next week, though; I'm on holidays right now :)

Bagira80 commented:
That is fine, thanks.
And enjoy your holidays.

adamwasila commented:
Hopefully, that should do the trick.

Bagira80 commented on src/main/resources/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder/config.jelly:
```diff
@@ -22,6 +22,9 @@
   <f:entry title="Number of builds today" help="/plugin/versionnumber/help-overrideNumbers.html">
     <f:textbox field="buildsToday" />
   </f:entry>
+  <f:entry title="Number of builds today" help="/plugin/versionnumber/help-overrideNumbers.html">
```
The title should obviously be "Number of builds this week".

Bagira80 commented on src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java:
```diff
@@ -316,6 +333,27 @@ private VersionNumberBuildInfo incBuild(AbstractBuild build, Map<String, String>
             }
             buildsToday = ((newVal >= 0) ? newVal : buildsToday);
         }
+        if (!this.oBuildsThisWeek.equals("")) {
```
Please change this line to the following:
`        if (this.oBuildsThisWeek == null || !this.oBuildsThisWeek.equals("")) {`

Bagira80 commented:
As there have been some changes to master lately, could you update your pull-request (again) with the latest changes from master?

Bagira80 commented:
After fixing the last mentioned typo, please rebase your branch, thereby
* squashing all your individual (and small) commits concerning `BUILD_WEEK` and `BUILD_THIS_WEEK` into a single commit,
* basing your new single commit on the last/current state of master branch of `versionnumber-plugin`.

For information about how to do this you can look [here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request).

Note: I could have just copied the changes of your individual commits and put them together in a single commit by myself, but then Git-log would mention me as the author and I did not want to steal the credits which belong to you. Doing it the rebase-way is just cleaner (and, of course, less work for me ;-) ).

adamwasila commented:
All right! I squashed and rebased all changes to a single commit then pushed it with -f flag to override master branch. Hopefully it is good enough to be merged now :)

Bagira80 commented:
Sorry for the late reply, I was on vacation and did not have much (stable) access to the internet.
Your pull-request is good to go, if you fix the tiny copy/paste error in the description field in `config.jelly`.
If you fix that and squash it into your single commit, we should be good to go and merge your pull-request.

Thanks again.
